packages/react-native-jigsaw/src/components/ToggleButton.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from "react";
+import { withTheme } from "../core/theming";
+import type { Theme } from "../styles/DefaultTheme";
+import { colorTypes } from "../types";
+import { StyleProp, StyleSheet, ViewStyle } from "react-native";
+import IconButton from "./IconButton";
+import {
+  COMPONENT_TYPES,
+  createIconProp,
+  createBoolProp,
+  createColorProp,
+  createNumberProp,
+  createActionProp,
+  createFieldNameProp,
+  createIconSizeProp,
+} from "../core/component-types";
+
+type Props = {
+  icon: string;
+  toggled?: boolean;
+  onPress?: () => void;
+  disabled?: boolean;
+  color?: colorTypes;
+  colorSecondary?: colorTypes;
+  borderColor?: colorTypes;
+  iconSize?: number;
+  width?: number;
+  height?: number;
+  style?: StyleProp<ViewStyle>;
+  theme: Theme;
+};
+
+const ToggleButton: React.FC<Props> = ({
+  icon,
+  toggled = false,
+  onPress = () => {},
+  disabled = false,
+  color = "primary",
+  colorSecondary = "surface",
+  borderColor = "divider",
+  iconSize = 25,
+  width = 50,
+  height = 50,
+  theme: { colors },
+  style,
+  ...rest
+}) => {
+  return (
+    <IconButton
+      icon={icon}
+      size={iconSize}
+      color={toggled ? colors[color] : colors[colorSecondary]}
+      onPress={onPress}
+      disabled={disabled}
+      style={[
+        styles.mainContainer,
+        {
+          width,
+          height,
+          backgroundColor: toggled ? colors[colorSecondary] : colors[color],
+          borderColor: colors[borderColor],
+        },
+        style,
+      ]}
```

Comment:
`rest` is defined but not used: still need to add it down here (into IconButton):

```<IconButton allotherprops {...rest} />```